### PR TITLE
Fix cds datasource

### LINKED ISSUE #41: Test fails because ['sub_region', 'city'] not in dataset. What to do?
### Description

I wrote a test for the hdx_formatter. The test failed because the `check_dataset_format` checks the order of columns in the dataset. The dataset that's fetch does not have 'sub_region,city' columns and so it fails. 

@ManuelAlvarezC What should be done in this case? Do I create an empty column or should I remove the hard check for the two columns from the `check_dataset_format` function?

### LINKED ISSUE #44: Wrong "aggregate" value in Spain country rows (should be "country" instead of "state")
* Commit SHA:
commit c4af4d71d781ac9fd204713b41cd79650519497c (HEAD -> master, origin/master, origin/HEAD)
Merge: f23824d 3de7d90
Author: Manuel Alvarez Campo <manuel@pythiac.com>
Date:   Sat Apr 4 13:48:16 2020 +0200

    Merge pull request #35 from shaikh-raj/master

    Adding metadata for CDS datasource
* Python version:3.7
* Operating System:Windows
* Data source: cds

### Description
Wrong "aggregate" value in Spain country rows (should be "country" instead of "state")

Code I use to change in exploration phase the error:
`data_cds.loc[(data_cds["country"] == "Spain") & (data_cds["state"].isnull()), "aggregate"] = "country"`

By the way, despite the normal little mistakes; great job of the api!

### PR DESCRIPTION
**Pull request by ManuelAlvarezC**
# Description

Changes wrong values on aggregate columns for rows with `state` value but no value on `state`, now `region` column.
Add tests for dataset validator.
Fix open issue with the validator. 

Fixes #41 Fixes #44 

# Checklist:

- [x] I have read the [CONTRIBUTING GUIDE](https://coronawhy.github.io/task-geo/contributing.html) and made sure this Pull Request is compliant with it.
- [x] I have read the [DATA MODEL](https://coronawhy.github.io/task-geo/data_model.html) and [DATA SOURCES](https://coronawhy.github.io/task-geo/data_sources.html) and made sure this Pull Request is compliant with it.
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: task_geo/data_sources/covid/cds/cds_formatter.py [hunk 1]
```diff
@@ -12,12 +12,20 @@ def cds_formatter(df):
...
     assert df[pd.isnull(df.country)].empty
 
-    df = df.rename(columns={'growthFactor': 'growth_factor'})
+    df = df.rename(columns={
+        'growthFactor': 'growth_factor',
+        'state': 'region',
+        'county': 'sub_region'
+    })
     df = df.reindex(columns=[
-        'city', 'county', 'state', 'country', 'population', 'lat', 'long',
```
Comment (DanielRobertNicoud) [inline task_geo/data_sources/covid/cds/cds_formatter.py:17]: Date should come directly after sub_region

**Comment by DanielRobertNicoud** [comment]
In __main__.py the argument of the function should be country="Italy". As it is currently, it returns an empty dataset.

File: task_geo/data_sources/covid/cds/cds_formatter.py (hunk 1 above)
Comment (ManuelAlvarezC) [reply task_geo/data_sources/covid/cds/cds_formatter.py:17]: I agree that `population`should not be returned.
On the other hand, date should come after all geopolitical columns, it's specified in the [data model documentation.](https://coronawhy.github.io/task-geo/data_model.html#time-related-data).

Comment (DanielRobertNicoud) [reply task_geo/data_sources/covid/cds/cds_formatter.py:17]: Well then we need to change the data model doc. The primary keys should come at the beginning of the datast.

**Pull request merged**